### PR TITLE
gitlab: clean up modules and utils

### PR DESCRIPTION
**Pull request by nejch**
##### SUMMARY

This cleans up the gitlab modules a bit to consistently use snake_case conventions (which is used in the newer modules, core utils, and python-gitlab). No functional change, but it's a small first step toward de-duplicating some of the code in those modules. It might help with implementing things like https://github.com/ansible-collections/community.general/issues/3669 in other modules.

I'm not sure if you accept these style-only PRs but thought I'd try. In case the PR is a bit cumbersome to review I split it into these commits (I can also split the PR):
- b04c5639 gitlab: use snake_case consistently in variables
- 92e03bd5 gitlab: use snake_case consistently in methods and functions
- 10f9a127 gitlab: remove dead code in module_utils

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
gitlab

##### ADDITIONAL INFORMATION

Not sure where the `request()` in utils came from but it is not used. It looks like a copy of the bitbucket helper, or maybe was there before `python-gitlab` was introduced. Also I'm aware a lot of the code could still be de-duplicated from those modules but didn't want to start with a huge PR.

Also wasn't quite sure how to label this but there's no "chore"/"style" issue type :grinning:

### CODE REVIEW

**Comment by nejch** [comment]
Oops, pre-commit was happy so I thought linting was fine. I'll fix asap.

**Comment by felixfontein** [comment]
@nejch pre-commit only checks a tiny subset of the existing linting rules.

**Comment by felixfontein** [comment]
Can you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)? For such refactorings, it's best to use `minor_changes`.

**Comment by nejch** [comment]
> Can you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)? For such refactorings, it's best to use `minor_changes`.

Thanks @felixfontein, I wasn't quite sure where this fits. Added now! :)

File: changelogs/fragments/3694-gitlab-cleanup.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - gitlab - clean up modules and utils (https://github.com/ansible-collections/community.general/pull/3694)
```
Comment (russoz) [inline changelogs/fragments/3694-gitlab-cleanup.yml:2]:
```suggestion
  - gitlab - clean up modules and utils (https://github.com/ansible-collections/community.general/pull/3694).
```

Comment (russoz) [reply changelogs/fragments/3694-gitlab-cleanup.yml:2]: Other than that, LGTM

File: plugins/modules/source_control/gitlab/gitlab_hook.py [hunk 2]
```diff
@@ -211,7 +211,7 @@ def createOrUpdateHook(self, project, hook_url, options):
                 'token': options['token']})
```
Comment (sh0shin) [inline plugins/modules/source_control/gitlab/gitlab_hook.py:211]:
```suggestion
                'token': options['token'],
                })
```

File: plugins/modules/source_control/gitlab/gitlab_hook.py [hunk 3]
```diff
@@ -224,7 +224,7 @@ def createOrUpdateHook(self, project, hook_url, options):
                 'enable_ssl_verification': options['enable_ssl_verification'],
                 'token': options['token']})
```
Comment (sh0shin) [inline plugins/modules/source_control/gitlab/gitlab_hook.py:225]:
```suggestion
                'token': options['token'],
                })
```

File: plugins/modules/source_control/gitlab/gitlab_hook.py [hunk 4]
```diff
@@ -346,41 +346,41 @@ def main():
...
+                                             "push_events_branch_filter": push_events_branch_filter,
+                                             "issues_events": issues_events,
+                                             "merge_requests_events": merge_requests_events,
+                                             "tag_push_events": tag_push_events,
+                                             "note_events": note_events,
+                                             "job_events": job_events,
+                                             "pipeline_events": pipeline_events,
+                                             "wiki_page_events": wiki_page_events,
+                                             "enable_ssl_verification": enable_ssl_verification,
+                                             "token": hook_token}):
```
Comment (sh0shin) [inline plugins/modules/source_control/gitlab/gitlab_hook.py:379]:
```suggestion
                                             "token": hook_token,
                                             }):
```

File: plugins/modules/source_control/gitlab/gitlab_runner.py [hunk 5]
```diff
@@ -220,15 +220,15 @@ def createOrUpdateRunner(self, description, options):
                 'tag_list': options['tag_list']})
```
Comment (sh0shin) [inline plugins/modules/source_control/gitlab/gitlab_runner.py:220]:
```suggestion
                'tag_list': options['tag_list'],
                })
```

File: plugins/modules/source_control/gitlab/gitlab_runner.py [hunk 6]
```diff
@@ -220,15 +220,15 @@ def createOrUpdateRunner(self, description, options):
...
             changed = True
         else:
-            changed, runner = self.updateRunner(self.runnerObject, {
+            changed, runner = self.update_runner(self.runner_object, {
                 'active': options['active'],
                 'locked': options['locked'],
                 'run_untagged': options['run_untagged'],
                 'maximum_timeout': options['maximum_timeout'],
                 'access_level': options['access_level'],
                 'tag_list': options['tag_list']})
```
Comment (sh0shin) [inline plugins/modules/source_control/gitlab/gitlab_runner.py:229]:
```suggestion
                'tag_list': options['tag_list'],
                })
```

File: plugins/modules/source_control/gitlab/gitlab_runner.py [hunk 7]
```diff
@@ -378,28 +378,28 @@ def main():
...
-                                              "registration_token": registration_token}):
-            module.exit_json(changed=True, runner=gitlab_runner.runnerObject._attrs,
+        if gitlab_runner.create_or_update_runner(runner_description, {
+                                                 "active": runner_active,
+                                                 "tag_list": tag_list,
+                                                 "run_untagged": run_untagged,
+                                                 "locked": runner_locked,
+                                                 "access_level": access_level,
+                                                 "maximum_timeout": maximum_timeout,
+                                                 "registration_token": registration_token}):
```
Comment (sh0shin) [inline plugins/modules/source_control/gitlab/gitlab_runner.py:398]:
```suggestion
                                                 "registration_token": registration_token,
                                                 }):
```

File: plugins/modules/source_control/gitlab/gitlab_user.py [hunk 8]
```diff
@@ -636,55 +636,55 @@ def main():
...
+                                             "sshkey_name": user_sshkey_name,
+                                             "sshkey_file": user_sshkey_file,
+                                             "sshkey_expires_at": user_sshkey_expires_at,
+                                             "group_path": group_path,
+                                             "access_level": access_level,
+                                             "confirm": confirm,
+                                             "isadmin": user_isadmin,
+                                             "external": user_external,
+                                             "identities": user_identities,
+                                             "overwrite_identities": overwrite_identities}):
```
Comment (sh0shin) [inline plugins/modules/source_control/gitlab/gitlab_user.py:684]:
```suggestion
                                             "overwrite_identities": overwrite_identities,
                                             }):
```

File: tests/unit/plugins/modules/source_control/gitlab/test_gitlab_group.py [hunk 9]
```diff
@@ -54,24 +54,24 @@ def setUp(self):
...
-                                             'description': "An interesting group",
-                                             'project_creation_level': "developer",
-                                             'subgroup_creation_level': "maintainer",
-                                             'require_two_factor_authentication': True})
+        group = self.moduleUtil.create_group({'name': "Foobar Group",
+                                              'path': "foo-bar",
+                                              'description': "An interesting group",
+                                              'project_creation_level': "developer",
+                                              'subgroup_creation_level': "maintainer",
+                                              'require_two_factor_authentication': True})
```
Comment (sh0shin) [inline tests/unit/plugins/modules/source_control/gitlab/test_gitlab_group.py:74]:
```suggestion
                                              'require_two_factor_authentication': True,
                                              })
```

File: tests/unit/plugins/modules/source_control/gitlab/test_gitlab_group.py [hunk 10]
```diff
@@ -84,11 +84,11 @@ def test_create_group(self):
...
-        group = self.moduleUtil.createGroup({'name': "BarFoo Group",
-                                             'path': "bar-foo",
-                                             'parent_id': 1,
-                                             'project_creation_level': "noone",
-                                             'require_two_factor_authentication': True})
+        group = self.moduleUtil.create_group({'name': "BarFoo Group",
+                                              'path': "bar-foo",
+                                              'parent_id': 1,
+                                              'project_creation_level': "noone",
+                                              'require_two_factor_authentication': True})
```
Comment (sh0shin) [inline tests/unit/plugins/modules/source_control/gitlab/test_gitlab_group.py:91]:
```suggestion
                                              'require_two_factor_authentication': True,
                                              })
```

File: tests/unit/plugins/modules/source_control/gitlab/test_gitlab_group.py [hunk 11]
```diff
@@ -101,29 +101,29 @@ def test_create_subgroup(self):
...
     def test_update_group(self):
         group = self.gitlab_instance.groups.get(1)
-        changed, newGroup = self.moduleUtil.updateGroup(group, {'name': "BarFoo Group",
-                                                                'visibility': "private",
-                                                                'project_creation_level': "maintainer",
-                                                                'require_two_factor_authentication': True})
+        changed, newGroup = self.moduleUtil.update_group(group, {'name': "BarFoo Group",
+                                                                 'visibility': "private",
+                                                                 'project_creation_level': "maintainer",
+                                                                 'require_two_factor_authentication': True})
```
Comment (sh0shin) [inline tests/unit/plugins/modules/source_control/gitlab/test_gitlab_group.py:107]:
```suggestion
                                                                 'require_two_factor_authentication': True,
                                                                 })
```

**Review by sh0shin — COMMENTED** [review]
Since we're into this, it would be nice to have.
Would result in less code deletion if we add future properties to the modules.

**Comment by nejch** [comment]
> Since we're into this, it would be nice to have. Would result in less code deletion if we add future properties to the modules.

Makes sense thanks @sh0shin! I was also thinking it might make sense to use a hanging indent (https://www.python.org/dev/peps/pep-0008/#indentation) rather than align with the function name - this way, if there is later a refactor again to use e.g. a mixin with `create()` for all resources or something, it would avoid another whitespace diff. WDYT?

It's already used in some places and that's where pep8 didn't complain after my change:
https://github.com/ansible-collections/community.general/blob/18a17acaa4647c96621e782099ad552bad646fb8/plugins/modules/source_control/gitlab/gitlab_runner.py#L212-L220

**Comment by felixfontein** [comment]
I personally prefer to have function arguments indented by the standard indent (4 spaces), instead of by something dependent on the function name (i.e. what @nejch is suggesting; I'm not sure that 'hanging indent' implies the 4 spaces, I think it also would allow the function name length dependent indent). Not just because it's easier to rename the function then, also because it's more uniform. But then I'm not maintaining this module, so feel free to ignore my voice on this :)

**Comment by felixfontein** [comment]
(I also like the trailing comma a lot, as it makes diffs less messy when adding more arguments.)

**Comment by russoz** [comment]
I am OK with that indentation either way. Still LGTM after the pending changes are applied.

**Comment by nejch** [comment]
I just applied the GUI suggestions but needs a little fixup locally, I'll push again in a moment.

**Review by felixfontein — APPROVED** [review]
Looks good to me!

**Review by russoz — APPROVED** [review]
LGTM! @sh0shin any last thing? Otherwise I believe we are ready to merge.

**Comment by sh0shin** [comment]
LGTM! Looks good!
Thanks @nejch for all the hard work!

**Review by sh0shin — APPROVED** [review]

**Comment by felixfontein** [comment]
@nejch thanks a lot for doing this!
@sh0shin @russoz thanks for reviewing!

**Pull request merged**